sunpy/net/fido_factory.py

```diff
@@ -95,12 +99,15 @@ def _handle_record_slice(self, client_resp, record_slice):
             resp = [client_resp[record_slice]]
         else:
             resp = client_resp[record_slice]
-
         # Reconstruct a response object with the sub-indexed records.
         ret = resp_type(resp)
         # Make sure we pass the client back out again.
         ret.client = client_resp.client
 
+        warnings.warn("Downloading of sliced results is not supported. "
```

Comment:
This warning will be triggered for all clients, even ones which do support this. This should be in `JSOCClient.__getitem__`.

Reply:
What clients support this?

Reply:
All of them but JSOC.